mesonbuild/interpreterbase.py

```diff
@@ -430,17 +444,18 @@ def evaluate_codeblock(self, node, start=0, end=None):
                 self.evaluate_statement(cur)
             except Exception as e:
                 if not hasattr(e, 'lineno'):
-                    e.lineno = cur.lineno
-                    e.colno = cur.colno
-                    e.file = os.path.join(self.subdir, 'meson.build')
+                    # We are doing the equivalent to setattr here and mypy does not like it
```

Comment:
you can make mypy happy by adding asserts I think.

Reply:
Mypy really, really doesn't like `hasattr` (apparently Van Rossum dislikes `hasattr`).

Reply:
I have tried, but I couldn't get anything besides `setattr` and `# type: ignore`. I went for `# type: ignore` because flake8 doesn't like `setattr`.